### ISSUE TITLE
Refactor Query class

### ISSUE THREAD

**Issue by greenape** [Flowminder/FlowKit#173]
The Query class is quite bloated. It needs to be refactored in a few ways - 
- Any methods which aren't actually useful should be removed
- Any methods which could be pushed out to functions taking a Query object should be pushed out
- The class should not depend on having a `Connection` object available as an attribute to function, and any methods which commune with the DB should take a connection as an argument instead

(I think this probably wants to be broken down into some sub issues.)

**Comment by greenape** [comment]
I think my ideal scenario here would really be that `Query` has just _one_ method, `make_query`, and everything else is either removed, a function, or a method on `Connection`, but I'd welcome some discussion on that.

**Comment by maxalbert** [comment]
Will need to take a closer look to make specific suggestions, but in general I suggest breaking this down into (relatively) small individual PRs which tackle one of the existing methods at a time. We should start with refactoring/deleting the ones where it's obvious that they are either not needed or in the wrong place. If there is anything left this will make it a lot clearer whether it really belongs in `Query`, or in `Connection` or in a separate function or even in a separate (not yet existing) class.

**Comment by greenape** [comment]
Yeah, I agree - I think a decent workflow for this is to do this discussion in this issue, and spin off into separate issues.

**Comment by maxalbert** [comment]
Yup, and also happy to leverage individual PR's for discussions on specific items. It can be quite useful to discuss things in work-in-progress PR's as they evolve, if we just want to explore where a particular refactoring might lead us.

**Comment by greenape** [comment]
Suggest we start by removing `__iter__`, `__next__` and `dependencies`, and shifting `dependency_graph`, `_adjacent`, `_get_deps` and `get_stored` out of Query.

Also suggest we rename `_adjacent` to something like `get_direct_dependencies` and use it where the `dependencies` prop is currently used, and `_get_dep` to `get_cached_dependencies`.
